common/djangoapps/student/models.py

```diff
@@ -3116,6 +3122,135 @@ class AccountRecoveryConfiguration(ConfigurationModel):
+        user_timezone = timezone(user_timezone_locale['user_timezone'] or str(UTC))
+        return user_timezone.localize(datetime.now())
+
+    @classmethod
+    def _calculate_streak_updates(cls, celebration, now):
+        """ Calculate the updates that should me applied to the streak fields of the provided celebration """
+        first_day_of_streak = celebration.first_day_of_streak
+        last_day_of_streak = celebration.last_day_of_streak
+        last_streak_celebration = celebration.last_streak_celebration
+        today = now.date()
+        yesterday = today - timedelta(days=1)
+        should_celebrate = False
+
+        first_ever_streak = None is last_streak_celebration is first_day_of_streak is last_day_of_streak
+        in_active_streak = last_day_of_streak and first_day_of_streak
+        broken_streak = last_day_of_streak not in (yesterday, today)
+        can_start_new_streak = last_day_of_streak and last_day_of_streak + timedelta(days=STREAK_BREAK_LENGTH) < today
+
+        if first_ever_streak:
+            # Start new streak
+            first_day_of_streak = today
+            last_day_of_streak = today
+        elif broken_streak:
+            if can_start_new_streak:
+                # If we were on a break and the break is over start new streak
+                first_day_of_streak = today
+                last_day_of_streak = today
+        elif in_active_streak:
+            # Increment existing streak
+            last_day_of_streak = today
+
+            streak_length = (last_day_of_streak - first_day_of_streak).days + 1
+            if streak_length == STREAK_LENGTH_TO_CELEBRATE:
+                # Celebrate if we didn't already celebrate today
+                if not last_streak_celebration == today:
+                    last_streak_celebration = today
+                    should_celebrate = True
+
+        return (first_day_of_streak, last_day_of_streak, last_streak_celebration), should_celebrate
+
+    @classmethod
+    def _update_streak(cls, celebration, streak_dates):
+        """ Update the celebration with the new streak data """
+        (first_day_of_streak, last_day_of_streak, last_streak_celebration) = streak_dates
+        # If anything needs to be updated, update the celebration in the database
+        if (first_day_of_streak != celebration.first_day_of_streak or
+                last_day_of_streak != celebration.last_day_of_streak or
+                last_streak_celebration != celebration.last_streak_celebration):
+            celebration.first_day_of_streak = first_day_of_streak
+            celebration.last_day_of_streak = last_day_of_streak
+            celebration.last_streak_celebration = last_streak_celebration
+            streak_length = (last_day_of_streak - first_day_of_streak).days + 1
+            if celebration.longest_ever_streak < streak_length:
+                celebration.longest_ever_streak = streak_length
+
+            celebration.save()
+        return celebration
```

Comment:
You never use this return, and callers already have this object because they pass it in. I'd drop it (or have a bool return value like whether you saved it or not or something, but if you aren't using it, I'd drop it).